soc/nordic/common/uicr/gen_uicr.py

```diff
@@ -198,6 +207,78 @@ class Uicr(c.LittleEndianStructure):
     ]
 
 
+def validate_secure_storage_partitions(args: argparse.Namespace) -> None:
+    """
+    Validate that secure storage partitions are laid out correctly.
+
+    Args:
+        args: Parsed command line arguments containing partition information
+
+    Raises:
+        ScriptError: If validation fails
+    """
+    # Expected order: cpuapp_crypto_partition, cpurad_crypto_partition,
+    # cpuapp_its_partition, cpurad_its_partition
+    partitions = [
+        PartitionInfo(
+            args.cpuapp_crypto_address, args.cpuapp_crypto_size, "cpuapp_crypto_partition"
+        ),
+        PartitionInfo(
+            args.cpurad_crypto_address, args.cpurad_crypto_size, "cpurad_crypto_partition"
+        ),
+        PartitionInfo(args.cpuapp_its_address, args.cpuapp_its_size, "cpuapp_its_partition"),
+        PartitionInfo(args.cpurad_its_address, args.cpurad_its_size, "cpurad_its_partition"),
+    ]
+
+    # Filter out zero-sized partitions (missing partitions)
+    present_partitions = [p for p in partitions if p.size > 0]
+
+    if not present_partitions:
+        # No partitions present - this is valid
+        return
+
+    # Check that the first present partition starts at the secure storage address
+    first_partition = present_partitions[0]
+    if first_partition.address != args.securestorage_address:
+        raise ScriptError(
+            f"First partition {first_partition.name} starts at {first_partition.address}, "
+            f"but must start at secure storage address {args.securestorage_address}"
+        )
```

Comment:
The start address and combined size of the partitions must also be aligned to 4 kB, or else you will get a boot error. I think it is worth adding checks for that as well, so that we catch that at build time instead.

We could maybe require that the combined size of the subpartitions is equal to `secure_storage_partition`, to ensure that the definition is coherent?

I think this part can also be dropped, which might make it easier to implement those other checks:
```python
if not present_partitions:
    # No partitions present - this is valid
    return
```
Unless I'm missing something, if you have set up the devicetree in that way (i.e. defined `secure_storage_partition` without any subpartitions) then you have probably made some sort of error.